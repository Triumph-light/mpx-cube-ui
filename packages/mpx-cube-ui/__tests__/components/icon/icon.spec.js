const simulate = require('@mpxjs/miniprogram-simulate')

describe('component icon unit test', function() {
  const componentId = simulate.loadMpx('src/components/icon/index.mpx')

  beforeEach(() => {
    jest.resetModules()
  })

  async function snapTest(options) {
    const component = simulate.render(componentId, options)
    component.attach(document.createElement('parent'))
    await simulate.sleep(0)
    expect(component.dom.innerHTML).toMatchSnapshot()
  }

  it('render', async () => {
    await snapTest({
      type: 'add'
    })
  })

  it('prop type', () => {
    const component = simulate.render(componentId, {
      type: 'add'
    })
    component.attach(document.createElement('parent'))
    const data = component.instance.data
    expect(data.type).toBe('add')
  })
})
