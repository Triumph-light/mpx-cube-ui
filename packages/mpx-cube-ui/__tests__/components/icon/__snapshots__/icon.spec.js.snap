// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component icon unit test render 1`] = `"<wx-view class=\\"main--cubeic main--cubeic main--cubeic-add\\" style=\\"\\"></wx-view>"`;
